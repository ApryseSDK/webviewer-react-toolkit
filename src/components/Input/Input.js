'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Input = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var hooks_1 = require('../../hooks');
var Icon_1 = require('../Icon');
exports.Input = (0, react_1.forwardRef)(function(_a, ref) {
  var _b = _a.message,
    message = _b === void 0 ? 'default' : _b,
    messageText = _a.messageText,
    fillWidth = _a.fillWidth,
    wrapperClassName = _a.wrapperClassName,
    padMessageText = _a.padMessageText,
    className = _a.className,
    onFocus = _a.onFocus,
    onBlur = _a.onBlur,
    rightElement = _a.rightElement,
    leftElement = _a.leftElement,
    _c = _a.type,
    type = _c === void 0 ? 'text' : _c,
    props = tslib_1.__rest(_a, [
      'message',
      'messageText',
      'fillWidth',
      'wrapperClassName',
      'padMessageText',
      'className',
      'onFocus',
      'onBlur',
      'rightElement',
      'leftElement',
      'type',
    ]);
  var _d = (0, hooks_1.useFocus)(onFocus, onBlur),
    focused = _d.focused,
    handleOnFocus = _d.handleOnFocus,
    handleOnBlur = _d.handleOnBlur;
  var rightIcon = (0, react_1.useMemo)(
    function() {
      if (rightElement) return rightElement;
      var icon = undefined;
      switch (message) {
        case 'warning':
          icon = 'Warning';
          break;
        case 'error':
          icon = 'Error';
          break;
      }
      return icon
        ? react_1.default.createElement(Icon_1.Icon, {
            className: 'ui__input__icon',
            icon: icon,
          })
        : undefined;
    },
    [message, rightElement],
  );
  var wrapperClass = (0, classnames_1.default)(
    'ui__base ui__input__wrapper',
    {
      'ui__input__wrapper--fill': fillWidth,
      'ui__input__wrapper--pad': padMessageText && !messageText,
    },
    wrapperClassName,
  );
  var mainClass = (0, classnames_1.default)(
    'ui__input',
    'ui__input--message-'.concat(message),
    { 'ui__input--focused': focused },
  );
  var inputClass = (0, classnames_1.default)(
    'ui__input__input',
    { 'ui__input__input--disabled': props.disabled },
    className,
  );
  return react_1.default.createElement(
    'div',
    { className: wrapperClass },
    react_1.default.createElement(
      'div',
      { className: mainClass },
      leftElement,
      react_1.default.createElement(
        'input',
        tslib_1.__assign({}, props, {
          type: type,
          onFocus: handleOnFocus,
          onBlur: handleOnBlur,
          className: inputClass,
          ref: ref,
        }),
      ),
      rightIcon,
    ),
    messageText
      ? react_1.default.createElement(
          'div',
          { className: 'ui__input__messageText' },
          messageText,
        )
      : undefined,
  );
});
//# sourceMappingURL=Input.js.map
