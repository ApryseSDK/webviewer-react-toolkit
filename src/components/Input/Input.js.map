{"version":3,"file":"Input.js","sourceRoot":"","sources":["Input.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAAmF;AACnF,qCAAuC;AACvC,gCAA+C;AA4DlC,QAAA,KAAK,GAAG,IAAA,kBAAU,EAC7B,UACE,EAaC,EACD,GAAG;IAbD,IAAA,eAAmB,EAAnB,OAAO,mBAAG,SAAS,KAAA,EACnB,WAAW,iBAAA,EACX,SAAS,eAAA,EACT,gBAAgB,sBAAA,EAChB,cAAc,oBAAA,EACd,SAAS,eAAA,EACT,OAAO,aAAA,EACP,MAAM,YAAA,EACN,YAAY,kBAAA,EACZ,WAAW,iBAAA,EACX,YAAa,EAAb,IAAI,mBAAG,MAAM,KAAA,EACV,KAAK,sBAZV,sJAaC,CADS;IAIJ,IAAA,KAA2C,IAAA,gBAAQ,EAAC,OAAO,EAAE,MAAM,CAAC,EAAlE,OAAO,aAAA,EAAE,aAAa,mBAAA,EAAE,YAAY,kBAA8B,CAAC;IAE3E,IAAM,SAAS,GAAG,IAAA,eAAO,EAAC;QACxB,IAAI,YAAY;YAAE,OAAO,YAAY,CAAC;QAEtC,IAAI,IAAI,GAA+B,SAAS,CAAC;QACjD,QAAQ,OAAO,EAAE;YACf,KAAK,SAAS;gBACZ,IAAI,GAAG,SAAS,CAAC;gBACjB,MAAM;YACR,KAAK,OAAO;gBACV,IAAI,GAAG,OAAO,CAAC;gBACf,MAAM;SACT;QAED,OAAO,IAAI,CAAC,CAAC,CAAC,8BAAC,WAAI,IAAC,SAAS,EAAC,iBAAiB,EAAC,IAAI,EAAE,IAAI,GAAI,CAAC,CAAC,CAAC,SAAS,CAAC;IAC7E,CAAC,EAAE,CAAC,OAAO,EAAE,YAAY,CAAC,CAAC,CAAC;IAE5B,IAAM,YAAY,GAAG,IAAA,oBAAU,EAC7B,6BAA6B,EAC7B;QACE,0BAA0B,EAAE,SAAS;QACrC,yBAAyB,EAAE,cAAc,IAAI,CAAC,WAAW;KAC1D,EACD,gBAAgB,CACjB,CAAC;IACF,IAAM,SAAS,GAAG,IAAA,oBAAU,EAAC,WAAW,EAAE,6BAAsB,OAAO,CAAE,EAAE,EAAE,oBAAoB,EAAE,OAAO,EAAE,CAAC,CAAC;IAC9G,IAAM,UAAU,GAAG,IAAA,oBAAU,EAAC,kBAAkB,EAAE,EAAE,4BAA4B,EAAE,KAAK,CAAC,QAAQ,EAAE,EAAE,SAAS,CAAC,CAAC;IAE/G,OAAO,CACL,uCAAK,SAAS,EAAE,YAAY;QAC1B,uCAAK,SAAS,EAAE,SAAS;YACtB,WAAW;YACZ,4DACM,KAAK,IACT,IAAI,EAAE,IAAI,EACV,OAAO,EAAE,aAAa,EACtB,MAAM,EAAE,YAAY,EACpB,SAAS,EAAE,UAAU,EACrB,GAAG,EAAE,GAAG,IACR;YACD,SAAS,CACN;QACL,WAAW,CAAC,CAAC,CAAC,uCAAK,SAAS,EAAC,wBAAwB,IAAE,WAAW,CAAO,CAAC,CAAC,CAAC,SAAS,CAClF,CACP,CAAC;AACJ,CAAC,CACF,CAAC"}