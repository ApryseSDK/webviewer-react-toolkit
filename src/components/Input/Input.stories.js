'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.PasswordExample = exports.SearchExample = exports.Controlled = exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importStar(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var Icon_1 = require('../Icon');
var IconButton_1 = require('../IconButton');
var Input_1 = require('../Input');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
var icons_1 = require('../../icons');
exports.default = {
  title: 'Components/Input',
  component: Input_1.Input,
  parameters: { readme: README_md_1.default },
};
var getTypeSelect = function() {
  return (0, addon_knobs_1.select)(
    'type',
    [
      'color',
      'date',
      'datetime-local',
      'email',
      // 'file',
      'month',
      'number',
      'password',
      // 'range',
      'search',
      'tel',
      'text',
      'time',
      'url',
      'week',
    ],
    'text',
  );
};
var Basic = function() {
  return react_1.default.createElement(Input_1.Input, {
    type: getTypeSelect(),
    message: (0, addon_knobs_1.select)(
      'message',
      ['default', 'warning', 'error'],
      'default',
    ),
    messageText: (0, addon_knobs_1.text)('messageText', ''),
    fillWidth: (0, addon_knobs_1.boolean)('fillWidth', false),
    padMessageText: (0, addon_knobs_1.boolean)('padMessageText', false),
    disabled: (0, addon_knobs_1.boolean)('disabled', false),
    onChange: (0, action_1.action)('onChange'),
  });
};
exports.Basic = Basic;
var Controlled = function() {
  return react_1.default.createElement(Input_1.Input, {
    value: (0, addon_knobs_1.text)('value', ''),
    type: getTypeSelect(),
    message: (0, addon_knobs_1.select)(
      'message',
      ['default', 'warning', 'error'],
      'default',
    ),
    messageText: (0, addon_knobs_1.text)('messageText', ''),
    fillWidth: (0, addon_knobs_1.boolean)('fillWidth', false),
    padMessageText: (0, addon_knobs_1.boolean)('padMessageText', false),
    disabled: (0, addon_knobs_1.boolean)('disabled', false),
    onChange: (0, action_1.action)('onChange'),
  });
};
exports.Controlled = Controlled;
var SearchExample = function() {
  return react_1.default.createElement(Input_1.Input, {
    type: 'search',
    leftElement: react_1.default.createElement(icons_1.Search, null),
  });
};
exports.SearchExample = SearchExample;
var PasswordExample = function() {
  var _a = (0, react_1.useState)(false),
    showPassword = _a[0],
    setShowPassword = _a[1];
  return react_1.default.createElement(Input_1.Input, {
    type: showPassword ? 'text' : 'password',
    message: (0, addon_knobs_1.select)(
      'message',
      ['default', 'warning', 'error'],
      'default',
    ),
    messageText: (0, addon_knobs_1.text)('messageText', ''),
    rightElement: react_1.default.createElement(
      IconButton_1.IconButton,
      {
        'aria-label': showPassword ? 'Hide password' : 'Show password',
        onClick: function() {
          return setShowPassword(function(s) {
            return !s;
          });
        },
      },
      react_1.default.createElement(Icon_1.Icon, {
        icon: showPassword ? 'Hide' : 'Show',
      }),
    ),
  });
};
exports.PasswordExample = PasswordExample;
//# sourceMappingURL=Input.stories.js.map
