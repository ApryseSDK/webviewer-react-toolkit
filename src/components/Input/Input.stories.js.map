{"version":3,"file":"Input.stories.js","sourceRoot":"","sources":["Input.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAA+D;AAC/D,qDAAwC;AACxC,gEAA+D;AAC/D,gCAA+B;AAC/B,4CAA2C;AAC3C,kCAAiC;AACjC,kEAAiC;AACjC,qCAAqC;AAErC,kBAAe,EAAE,KAAK,EAAE,kBAAkB,EAAE,SAAS,EAAE,aAAK,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAEvF,IAAM,aAAa,GAAG;IACpB,OAAO,IAAA,oBAAM,EACX,MAAM,EACN;QACE,OAAO;QACP,MAAM;QACN,gBAAgB;QAChB,OAAO;QACP,UAAU;QACV,OAAO;QACP,QAAQ;QACR,UAAU;QACV,WAAW;QACX,QAAQ;QACR,KAAK;QACL,MAAM;QACN,MAAM;QACN,KAAK;QACL,MAAM;KACP,EACD,MAAM,CACP,CAAC;AACJ,CAAC,CAAC;AAEK,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,8BAAC,aAAK,IACJ,IAAI,EAAE,aAAa,EAAE,EACrB,OAAO,EAAE,IAAA,oBAAM,EAAC,SAAS,EAAE,CAAC,SAAS,EAAE,SAAS,EAAE,OAAO,CAAC,EAAE,SAAS,CAAC,EACtE,WAAW,EAAE,IAAA,kBAAI,EAAC,aAAa,EAAE,EAAE,CAAC,EACpC,SAAS,EAAE,IAAA,qBAAO,EAAC,WAAW,EAAE,KAAK,CAAC,EACtC,cAAc,EAAE,IAAA,qBAAO,EAAC,gBAAgB,EAAE,KAAK,CAAC,EAChD,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,EACpC,QAAQ,EAAE,IAAA,eAAM,EAAC,UAAU,CAAC,GAC5B,CACH,EAV0B,CAU1B,CAAC;AAVW,QAAA,KAAK,SAUhB;AAEK,IAAM,UAAU,GAAG,cAAM,OAAA,CAC9B,8BAAC,aAAK,IACJ,KAAK,EAAE,IAAA,kBAAI,EAAC,OAAO,EAAE,EAAE,CAAC,EACxB,IAAI,EAAE,aAAa,EAAE,EACrB,OAAO,EAAE,IAAA,oBAAM,EAAC,SAAS,EAAE,CAAC,SAAS,EAAE,SAAS,EAAE,OAAO,CAAC,EAAE,SAAS,CAAC,EACtE,WAAW,EAAE,IAAA,kBAAI,EAAC,aAAa,EAAE,EAAE,CAAC,EACpC,SAAS,EAAE,IAAA,qBAAO,EAAC,WAAW,EAAE,KAAK,CAAC,EACtC,cAAc,EAAE,IAAA,qBAAO,EAAC,gBAAgB,EAAE,KAAK,CAAC,EAChD,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,EACpC,QAAQ,EAAE,IAAA,eAAM,EAAC,UAAU,CAAC,GAC5B,CACH,EAX+B,CAW/B,CAAC;AAXW,QAAA,UAAU,cAWrB;AAEK,IAAM,aAAa,GAAG,cAAM,OAAA,8BAAC,aAAK,IAAC,IAAI,EAAC,QAAQ,EAAC,WAAW,EAAE,8BAAC,cAAM,OAAG,GAAI,EAAhD,CAAgD,CAAC;AAAvE,QAAA,aAAa,iBAA0D;AAE7E,IAAM,eAAe,GAAG;IACvB,IAAA,KAAkC,IAAA,gBAAQ,EAAC,KAAK,CAAC,EAAhD,YAAY,QAAA,EAAE,eAAe,QAAmB,CAAC;IAExD,OAAO,CACL,8BAAC,aAAK,IACJ,IAAI,EAAE,YAAY,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,UAAU,EACxC,OAAO,EAAE,IAAA,oBAAM,EAAC,SAAS,EAAE,CAAC,SAAS,EAAE,SAAS,EAAE,OAAO,CAAC,EAAE,SAAS,CAAC,EACtE,WAAW,EAAE,IAAA,kBAAI,EAAC,aAAa,EAAE,EAAE,CAAC,EACpC,YAAY,EACV,8BAAC,uBAAU,kBACG,YAAY,CAAC,CAAC,CAAC,eAAe,CAAC,CAAC,CAAC,eAAe,EAC5D,OAAO,EAAE,cAAM,OAAA,eAAe,CAAC,UAAC,CAAC,IAAK,OAAA,CAAC,CAAC,EAAF,CAAE,CAAC,EAA1B,CAA0B;YAEzC,8BAAC,WAAI,IAAC,IAAI,EAAE,YAAY,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,MAAM,GAAI,CACnC,GAEf,CACH,CAAC;AACJ,CAAC,CAAC;AAlBW,QAAA,eAAe,mBAkB1B"}