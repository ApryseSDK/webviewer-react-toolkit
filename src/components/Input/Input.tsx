import classnames from 'classnames';
import React, { forwardRef, InputHTMLAttributes, ReactNode, useMemo } from 'react';
import { useFocus } from '../../hooks';
import { AvailableIcons, Icon } from '../Icon';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  /**
   * Messaging feedback for the input. This can be used for displaying feedback
   * inside of forms.
   * @default "default"
   */
  message?: 'default' | 'warning' | 'error';
  /**
   * Text to display under the input describing the message.
   */
  messageText?: string;
  /**
   * Should this input fill the width of the parent container.
   */
  fillWidth?: boolean;
  /**
   * A class for the wrapper div which encompases the input and icon and warning
   * texrt. Use this to do things like add margin-bottom to the entire input for
   * alignment in forms.
   */
  wrapperClassName?: string;
  /**
   * If true, will add padding below the input to compensate for message text
   * being absent. This prevents UI jumping when message text appears.
   */
  padMessageText?: boolean;
  /**
   * If provided will wrap the icon in an icon button, allowing handling of
   * onClick events and attaching aria-labels. This will override any default
   * icons like the ones for warning and error messages.
   */
  rightElement?: ReactNode;
}

export const Input = forwardRef<HTMLInputElement, InputProps>(
  (
    {
      message = 'default',
      messageText,
      fillWidth,
      wrapperClassName,
      padMessageText,
      className,
      onFocus,
      onBlur,
      rightElement,
      ...props
    },
    ref,
  ) => {
    const { focused, handleOnFocus, handleOnBlur } = useFocus(onFocus, onBlur);

    const rightIcon = useMemo(() => {
      if (rightElement) return rightElement;

      let icon: AvailableIcons | undefined = undefined;
      switch (message) {
        case 'warning':
          icon = 'Warning';
          break;
        case 'error':
          icon = 'Error';
          break;
      }

      return icon ? <Icon className="ui__input__icon" icon={icon} /> : undefined;
    }, [message, rightElement]);

    const wrapperClass = classnames(
      'ui__base ui__input__wrapper',
      {
        'ui__input__wrapper--fill': fillWidth,
        'ui__input__wrapper--pad': padMessageText && !messageText,
      },
      wrapperClassName,
    );
    const mainClass = classnames('ui__input', `ui__input--message-${message}`, { 'ui__input--focused': focused });
    const inputClass = classnames('ui__input__input', { 'ui__input__input--disabled': props.disabled }, className);

    return (
      <div className={wrapperClass}>
        <div className={mainClass}>
          <input {...props} onFocus={handleOnFocus} onBlur={handleOnBlur} className={inputClass} ref={ref} />
          {rightIcon}
        </div>
        {messageText ? <div className="ui__input__messageText">{messageText}</div> : undefined}
      </div>
    );
  },
);
