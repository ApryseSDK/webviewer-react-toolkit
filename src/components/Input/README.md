A simple styled input with the ability to show error or warning feedback.
