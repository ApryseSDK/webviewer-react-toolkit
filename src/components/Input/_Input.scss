.ui__input__wrapper {
  width: 200px;

  &--fill {
    width: 100%;
  }

  &--pad {
    padding-bottom: 20px;
  }
}

.ui__input {
  border-radius: $border-radius;
  transition: $focus-transition;
  background-color: $color-gray-1;
  border: 1px solid $color-gray-3;
  display: inline-flex;
  width: 100%;

  &--message {
    &-default {
      &.ui__input--focused {
        border: 1px solid $color-theme-primary;
      }

      &.ui__input--focused {
        @include focus;
      }
    }

    &-warning {
      border-color: $color-message-warning;

      .ui__input__icon {
        fill: $color-message-warning;
      }

      &.ui__input--focused {
        box-shadow: 0 0 0 2px var(--color-message-warning-focus-shadow);
      }
    }

    &-error {
      border-color: $color-message-error;

      .ui__input__icon {
        fill: $color-message-error;
      }

      &.ui__input--focused {
        box-shadow: 0 0 0 2px var(--color-message-error-focus-shadow);
      }
    }
  }

  .ui__input__input {
    flex: 1 1 auto;
    padding: $padding-tiny;
    height: 28px;
    outline: none;
    border: none;
    background-color: transparent;
    color: $color-font-primary;
    min-height: 24px;

    &--disabled {
      opacity: 0.7;
    }
  }

  &__icon {
    flex: 0 0;
    padding: 0 2px;
  }

  &__messageText {
    width: 100%;
    font-size: $font-size-small;
    margin-top: $padding-tiny;
  }
}
