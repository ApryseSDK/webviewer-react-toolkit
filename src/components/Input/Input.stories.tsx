import { boolean, select, text } from '@storybook/addon-knobs';
import React, { useState } from 'react';
import { action } from '../../storybook-helpers/action/action';
import { Icon } from '../Icon';
import { IconButton } from '../IconButton';
import { Input } from '../Input';
import readme from './README.md';
import { Search } from '../../icons';

export default { title: 'Components/Input', component: Input, parameters: { readme } };

const getTypeSelect = () => {
  return select(
    'type',
    [
      'color',
      'date',
      'datetime-local',
      'email',
      // 'file',
      'month',
      'number',
      'password',
      // 'range',
      'search',
      'tel',
      'text',
      'time',
      'url',
      'week',
    ],
    'text',
  );
};

export const Basic = () => (
  <Input
    type={getTypeSelect()}
    message={select('message', ['default', 'warning', 'error'], 'default')}
    messageText={text('messageText', '')}
    fillWidth={boolean('fillWidth', false)}
    padMessageText={boolean('padMessageText', false)}
    disabled={boolean('disabled', false)}
    onChange={action('onChange')}
  />
);

export const Controlled = () => (
  <Input
    value={text('value', '')}
    type={getTypeSelect()}
    message={select('message', ['default', 'warning', 'error'], 'default')}
    messageText={text('messageText', '')}
    fillWidth={boolean('fillWidth', false)}
    padMessageText={boolean('padMessageText', false)}
    disabled={boolean('disabled', false)}
    onChange={action('onChange')}
  />
);

export const SearchExample = () => <Input type="search" leftElement={<Search />} />;

export const PasswordExample = () => {
  const [showPassword, setShowPassword] = useState(false);

  return (
    <Input
      type={showPassword ? 'text' : 'password'}
      message={select('message', ['default', 'warning', 'error'], 'default')}
      messageText={text('messageText', '')}
      rightElement={
        <IconButton
          aria-label={showPassword ? 'Hide password' : 'Show password'}
          onClick={() => setShowPassword(s => !s)}
        >
          <Icon icon={showPassword ? 'Hide' : 'Show'} />
        </IconButton>
      }
    />
  );
};
