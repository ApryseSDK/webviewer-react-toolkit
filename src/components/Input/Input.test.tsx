import { shallow } from 'enzyme';
import React from 'react';
import { Input } from '../Input';

describe('Input component', () => {
  it('renders its contents', () => {
    const input = shallow(<Input />);
    expect(input.find('.ui__input')).toHaveLength(1);
  });

  it('snapshot renders default input', () => {
    const input = shallow(<Input />);
    expect(input).toMatchSnapshot();
  });
});
