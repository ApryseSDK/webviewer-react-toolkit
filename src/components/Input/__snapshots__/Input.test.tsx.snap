// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component snapshot renders default input 1`] = `
<div
  className="ui__base ui__input__wrapper"
>
  <div
    className="ui__input ui__input--message-default"
  >
    <input
      className="ui__input__input"
      onBlur={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;
